llvm/lib/Transforms/Scalar/TailRecursionElimination.cpp

```diff
@@ -510,7 +510,9 @@ void TailRecursionEliminator::createTailRecurseLoopHeader(CallInst *CI) {
   NewEntry->takeName(HeaderBB);
   HeaderBB->setName("tailrecurse");
   BranchInst *BI = BranchInst::Create(HeaderBB, NewEntry);
```

Comment:
BI is unused now (but referenced in the comment at line 515), leading to compiler warnings.
```
../lib/Transforms/Scalar/TailRecursionElimination.cpp:512:15: error: unused variable 'BI' [-Werror,-Wunused-variable]
  BranchInst *BI = BranchInst::Create(HeaderBB, NewEntry);
              ^
1 error generated.
```

Reply:
Thanks for pointing out this. I fix this in #86927.